SimG4Core/Generators/src/Generator.cc

```diff
@@ -244,6 +248,9 @@ void Generator::HepMC2G4(const HepMC::GenEvent *evt_orig, G4Event *g4evt) {
       if (status > 3 && isExotic(pdg) && (!(isExoticNonDetectable(pdg)))) {
         status = hasDecayVertex ? 2 : 1;
       }
+      if (status == 2 && abs(pdg) == 9900015) {
```

Comment:
@mgratti , why not to add following lines:

// heavy neutrino should not be propagated by Geant4
if(std::abs(pdg) == 9900015) status = 3;

Reply:
@civanch, sure, this clearly preserves my case.